projects/103_hw_megan/src/main.c

```diff
@@ -0,0 +1,64 @@
+#include "adc.h"
+#include "gpio.h"
+#include "gpio_it.h"
+#include "interrupt.h"
+#include "log.h"
+#include "wait.h"
+
+// logs the output of the ADC reading from pin A6
+static void prv_log_adc_reading(const GpioAddress *address, void *context) {
+  GpioAddress *adc_addr = context;
+  uint16_t adc_data = 0;
+  adc_read_converted_pin(*adc_addr, &adc_data);
+
+  LOG_DEBUG("The ADC converted reading is: %d\n", adc_data);
+}
+
+int main() {
+  interrupt_init();
+  gpio_init();
+
```

Comment:
`gpio_it` needs to be initialized as well since we are using interrupts with our GPIO pins